src/dbnode/persist/fs/seek_manager.go

```diff
@@ -437,6 +437,32 @@ func (m *seekerManager) Return(
 	return nil
 }
 
+func (m *seekerManager) ReturnShard(shard uint32) error {
+	if int(shard) >= len(m.seekersByShardIdx) {
+		return errShardNotExists
+	}
+	byTime := m.seekersByShardIdx[shard]
```

Comment:
I wonder if we should set `m.seekersByShardIdx[shard]` to `nil` afterwards?

Reply:
Other code paths in `seekManager` do not check for `nil` when retrieving `byTime` so by doing this we would brake some other code.

Reply:
Looking at this code, I think the assumption is that each element should be initialized

https://github.com/m3db/m3/blob/255a4b14df1da9b4bc702814933bd8b15d0b3caf/src/dbnode/persist/fs/seek_manager.go#L899-L906